app/scripts/controllers/transactions/lib/util.js

```diff
@@ -37,19 +38,30 @@ export function normalizeTxParams(txParams, lowerCase = true) {
  * @throws {Error} if the tx params contains invalid fields
  */
 export function validateTxParams(txParams) {
+  if (!txParams || typeof txParams !== 'object' || Array.isArray(txParams)) {
+    throw ethErrors.rpc.invalidParams(
+      'Invalid transaction params: must be an object.',
+    )
+  }
+  if (!txParams.to && !txParams.data) {
+    throw ethErrors.rpc.invalidParams(
+      'Invalid transaction params: must specify "to" and/or "data".',
```

Comment:
It might be helpful to explain which property is required in which scenario. Something like:

```suggestion
      'Invalid transaction params: must specify "data" for contract deployments, or "to" for all other types of transactions.',
```

Reply:
Good idea! I slightly amended your suggestion to account for the fact that both `to` and `data` can be provided: 07338ae